rustls/src/tls13/mod.rs

```diff
@@ -199,6 +209,76 @@ impl MessageDecrypter for Tls13MessageDecrypter {
     }
 }
 
+struct RingAeadFactory(&'static ring::aead::Algorithm);
```

Comment:
The `Factory` naming seems unidiomatic to me. Maybe `RingAeadBuilder`? Feels like these should just be moved into a `ring` module directly so we can then ditch the prefix.

Ordering: `AeadChacha20Poly1305` (and co), then `RingAeadFactory`, then `Tls13MessageEncrypter` (and `Decrypter`)?